libs/common-services/src/main/java/uk/gov/di/ipv/core/library/helpers/RequestHelper.java

```diff
@@ -263,27 +246,19 @@ private static void validateIpvSessionId(
         }
     }
 
-    private static String getIpAddress(Map<String, String> headers)
-            throws HttpResponseExceptionWithErrorBody {
+    private static String getIpAddress(Map<String, String> headers) {
         String ipAddress = RequestHelper.getHeaderByKey(headers, IP_ADDRESS_HEADER);
-        validateIpAddress(ipAddress, String.format("%s not present in header", IP_ADDRESS_HEADER));
+        validateIpAddress(ipAddress);
         return ipAddress;
     }
 
-    private static void validateIpAddress(String ipAddress, String errorMessage)
-            throws HttpResponseExceptionWithErrorBody {
+    private static void validateIpAddress(String ipAddress) {
         if (ipAddress == null) {
-            LOGGER.error(LogHelper.buildErrorMessage(errorMessage, IP_ADDRESS_HEADER));
-            throw new HttpResponseExceptionWithErrorBody(
-                    SC_BAD_REQUEST, ErrorResponse.MISSING_IP_ADDRESS);
+            LOGGER.warn(LogHelper.buildErrorMessage(ErrorResponse.MISSING_IP_ADDRESS));
         }
     }
 
-    private static void validateClientOAuthSessionId(
-            String clientSessionId, StringMapMessage message) {
-        if (clientSessionId == null) {
```

Comment:
There was a subtlety here - it would log a warning if the `clientSessionId` was null, but not if it was empty.

Because of the way our VTL works in API Gateway, it would never actually be null - if the header was missing it'd default to an empty string. I've added `nullIfBlank` to get the null we expect.

In practice, it will be missing in 99% of cases anyway, because we only use this for the cross-browser recovery - so we shouldn't bother logging a warning.